src/main.rs

```diff
@@ -366,6 +379,77 @@ impl Context {
     }
 }
 
+// Fetches remote podcast/episode names by guids using the Podcastindex API and caches results into an LRU cache
+pub async fn fetch_podcast_episode_by_guid(cache: &mut LruCache<String, Option<PodcastEpisodeGuid>>, podcast_guid: String, episode_guid: String) -> Option<PodcastEpisodeGuid> {
+    let key = format!("{}_{}", podcast_guid, episode_guid);
+
+    if let Some(cached_guid) = cache.get(&key) {
+        println!("Remote podcast/episode from cache: {:#?}", cached_guid);
+        return cached_guid.clone(); // already exists in cache
+    }
+
+    match fetch_api_podcast_episode_by_guid(&podcast_guid, &episode_guid).await {
+        Ok(Some(guid)) => {
+            println!("Remote podcast/episode from API: {:#?}", guid);
+            cache.put(key, Some(guid.clone())); // cache to avoid spamming api
+            Some(guid)
+        },
+        Ok(None) => {
+            println!("Remote podcast/episode not found {} {}", podcast_guid, episode_guid);
+            cache.put(key, None); // cache to avoid spamming api
+            None
+        }
+        Err(e) => {
+            eprintln!("Error retrieving remote podcast/episode from API: {:#?}", e);
+            None
+        }
+    }
+}
+
+// Fetches remote podcast/episode names by guids using the Podcastindex API
+pub async fn fetch_api_podcast_episode_by_guid(podcast_guid: &String, episode_guid: &String) -> Result<Option<PodcastEpisodeGuid>, Error> {
+    // using the auth-free value api to get feed and item titles
+    let url = format!(
+        "https://api.podcastindex.org/api/1.0/value/byepisodeguid?podcastguid={}&episodeguid={}",
+        podcast_guid,
+        episode_guid
+    );
```

Comment:
I should use the RequestBuilder to properly escape url params https://docs.rs/reqwest/0.8.6/reqwest/struct.RequestBuilder.html#method.query